indico/modules/vc/models/vc_rooms.py

```diff
@@ -316,32 +341,35 @@ def get_linked_for_event(cls, event):
         """Get a dict mapping link objects to event vc rooms."""
         return {vcr.link_object: vcr for vcr in cls.find_for_event(event)}
 
-    def delete(self, user, delete_all=False):
+    def delete(self, user: User, *, check_vc_room: bool = True):
         """Delete a VC room from an event.
 
         If the room is not used anywhere else, the room itself is also deleted.
 
         :param user: the user performing the deletion
-        :param delete_all: if True, the room is detached from all
-                           events and deleted.
+        :param check_vc_room: whether to check as well if the VCRoom can be deleted
+                              (no more associations left)
         """
+        # send signals
+        signals.vc.detached_vc_room.send(self, vc_room=self.vc_room, old_link=self.link_object, event=self.event)
+
+        Logger.get('modules.vc').info(
+            'Detaching videoconference %s from event %s (%s)', self.vc_room, self.event, self.link_object
+        )
+
+        # do the actual deletion from the DB
         vc_room = self.vc_room
-        if delete_all:
-            for assoc in vc_room.events[:]:
-                Logger.get('modules.vc').info('Detaching videoconference %s from event %s (%s)',
-                                              vc_room, assoc.event, assoc.link_object)
-                vc_room.events.remove(assoc)
-        else:
-            Logger.get('modules.vc').info('Detaching videoconference %s from event %s (%s)',
-                                          vc_room, self.event, self.link_object)
-            vc_room.events.remove(self)
+        self.vc_room.events.remove(self)
         db.session.flush()
-        if vc_room.plugin and not vc_room.events:
-            Logger.get('modules.vc').info(f'Deleting videoconference {vc_room}')
-            if vc_room.status != VCRoomStatus.deleted:
-                vc_room.plugin.delete_room(vc_room, self.event)
-                notify_deleted(vc_room.plugin, vc_room, self, self.event, user)
-            db.session.delete(vc_room)
+
+        # process plugin actions
+        if plugin := vc_room.plugin:
```

Comment:
Did you remove the `VCRoomStatus.deleted` check here on purpose? Not sure if it's useful to tell a plugin about detaching a deleted room... (but I guess there's also no harm in doing so)

Reply:
haven't checked. in any case the vcroom can get into the deleted state when you refresh the data from zoom and the meeting on longer exists there. probably likewise when receiving the webhook from zoom about it being deleted

Reply:
Indeed. Anyway, as you said, there is no harm in notifying the plugin, although there is probably no useful thing we'd need to to with it. But I'd keep it as it is in that case.